edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/WorkloadClient.cs

```diff
@@ -1,150 +1,160 @@
-            }
-
-            internal static void ExecutingOperation(string operation, string url)
-            {
-                Log.LogDebug((int)EventIds.ExecutingOperation, $"Making a Http call to {url} to {operation}");
-            }
-
-            internal static void SuccessfullyExecutedOperation(string operation, string url)
-            {
-                Log.LogDebug((int)EventIds.SuccessfullyExecutedOperation, $"Received a valid Http response from {url} for {operation}");
-            }
-        }
-    }
-}
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Util.Edged
+{
+    using System;
+    using System.Net.Http;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util.Edged.GeneratedCode;
+    using Microsoft.Azure.Devices.Edge.Util.TransientFaultHandling;
+    using Microsoft.Extensions.Logging;
+
+    public class WorkloadClient
+    {
+        static readonly ITransientErrorDetectionStrategy TransientErrorDetectionStrategy = new ErrorDetectionStrategy();
+        static readonly RetryStrategy TransientRetryStrategy =
+            new ExponentialBackoff(retryCount: 3, minBackoff: TimeSpan.FromSeconds(2), maxBackoff: TimeSpan.FromSeconds(30), deltaBackoff: TimeSpan.FromSeconds(3));
+        readonly Uri workloadUri;
+        readonly string apiVersion;
+        readonly string moduleId;
+        readonly string moduleGenerationId;
+
+        public WorkloadClient(Uri serverUri, string apiVersion, string moduleId, string moduleGenerationId)
+        {
+            this.workloadUri = Preconditions.CheckNotNull(serverUri, nameof(serverUri));
+            this.apiVersion = Preconditions.CheckNonWhiteSpace(apiVersion, nameof(apiVersion));
+            this.moduleId = Preconditions.CheckNonWhiteSpace(moduleId, nameof(moduleId));
+            this.moduleGenerationId = Preconditions.CheckNonWhiteSpace(moduleGenerationId, nameof(moduleGenerationId));
+        }
+
+        public async Task<CertificateResponse> CreateServerCertificateAsync(string hostname, DateTime expiration)
+        {
+            var request = new ServerCertificateRequest
+            {
+                CommonName = hostname,
+                Expiration = expiration
+            };
+
+            using (HttpClient httpClient = HttpClientHelper.GetHttpClient(this.workloadUri))
+            {
+                var edgeletHttpClient = new HttpWorkloadClient(httpClient) { BaseUrl = HttpClientHelper.GetBaseUrl(this.workloadUri) };
+                CertificateResponse result = await this.Execute(() => edgeletHttpClient.CreateServerCertificateAsync(this.apiVersion, this.moduleId, this.moduleGenerationId, request), "CreateServerCertificateAsync");
+                return result;
+            }
+        }
+
+        public async Task<TrustBundleResponse> GetTrustBundleAsync()

```

Comment:
Perhaps my editor setting caused the diff here but this function was the only change.